src/utils/networkHealth.ts

```diff
@@ -0,0 +1,89 @@
+import { NetworkHealth } from 'interfaces/networkHealth.interface'
+import { BlockTimeHealth } from '../interfaces/blockTimeHealth.interface'
+import { getLatestBlocks } from '../repositories/block.repository'
+import { logger } from './logger'
+import { networkHealthConfig } from './networkHealthConfig'
+import { networkHealthNotification } from './notifications'
+
+export async function checkNetworkHealth() {
+  const blockTimeHealth = await calculateBlockTimeHealth()
+  if (!blockTimeHealth) return
+
+  // will be expanded with additional checks
+  const networkHealth: NetworkHealth = {
+    blockTimeHealth
+  }
+  logger.info(JSON.stringify(networkHealth))
+
+  if (!blockTimeHealth.healthy) {
+    networkHealthNotification(networkHealth)
+  }
+}
+
+async function calculateBlockTimeHealth(): Promise<BlockTimeHealth> {
+  const {
+    allowedNumberDeviatedBlocks,
+    allowedSingleBlockTimeDeviationInSeconds,
+    maxAllowedAverageBlocktime,
+    minAllowedAverageBlocktime,
+    numberOfBlocks,
+    targetBlockTimeInSeconds
+  } = networkHealthConfig
+  const minAllowedBlocktime = targetBlockTimeInSeconds - allowedSingleBlockTimeDeviationInSeconds
+  const maxAllowedBlocktime = targetBlockTimeInSeconds + allowedSingleBlockTimeDeviationInSeconds
+  const minNumberOfBlocks = 10
```

Comment:
nitpick(blocking): Add a comment about why you use 10 blocks. It will make refactoring and future development easier.